lib/TAP.pm

```diff
@@ -806,17 +831,18 @@ class Async {
 
     multi get_runner(Source::Proc $proc) {
         my $async = Proc::Async.new($proc.path, $proc.args);
-        my $events = parser($async.stdout);
+        my $source;
         state $devnull;
         END { $devnull.close with $devnull }
         given $proc.err {
             my $err = $_;
             when 'stderr' { #default is correct
             }
             when 'merge' {
-                warn "Merging isn't supported yet on Asynchronous streams";
-                $devnull //= open($*SPEC.devnull, :w);
-                $async.bind-stderr($devnull);
+                $source = $async.Supply;
```

Comment:
This very very much should have been a separate commit. It's a rather big change that is unrelated to the rest of the patch.

As far as I know it's also wrong: AFAICT it's mixing the input in the asynchronous IO system, instead of making them one descriptor. This makes it unusable for our purposes.

I would very much love to be proben wrong here though.

Reply:
For so long I didn't see any scrambling related to use of `.Supply`. Perhaps I was just lucky.